smart-contract/contracts/YourNftToken.sol

```diff
@@ -80,18 +80,20 @@ contract YourNftToken is ERC721A, Ownable, ReentrancyGuard {
     address latestOwnerAddress;
 
     while (ownedTokenIndex < ownerTokenCount && currentTokenId <= maxSupply) {
-      TokenOwnership memory ownership = _ownerships[currentTokenId];
+      if (_exists(currentTokenId))
+      {
```

Comment:
Please follow the same coding style as the rest of the project: there shouldn't be a new line before this opening bracket.